### PR TITLE
feat(ts): move user module to ts

### PR DESCRIPTION
**Pull request by sibelius**
mongoose is properly typecheck now

- interface IUser is causing problems
- `as` type cast is not working either

GraphQLNonNull does not work without `new` syntax

we should update `graphql` typedefs

### CODE REVIEW

**Comment by sibelius** [comment]
related issues

https://github.com/babel/babel/issues/8663
https://github.com/babel/babel/issues/8664

we can workaround both of them for now

but it'd not make `schema.pre` typesafe

**Comment by sibelius** [comment]
merge on green

File: src/app.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
 // @flow
 import 'isomorphic-fetch';
 
-import Koa from 'koa';
+import Koa, { Request } from 'koa';
```
Comment (jgcmarins) [inline src/app.ts]: missing `type`

Comment (sibelius) [reply src/app.ts]: there is no need for `type` on `TS`

Comment (jgcmarins) [reply src/app.ts]: nice

**Pull request merged**